File: src/main/kotlin/org/avlasov/example/other/java/JavaAnnotations.java
package org.avlasov.example.other.java;

import java.io.IOException;
import java.security.Security;

/**
 * Created By artemvlasov on 2018-10-30
 **/
public class JavaAnnotations {

    public static void main(String[] args) {
        int compare = Key.COMPARATOR.compare(new Key(1), new Key(2));
        Singleton.provider = Security.getProvider("test");
        int aConst = Obj.CONST;
        int version = C.VERSION;
        int max = JavaAnnotationsKt.MAX;
        JmvStatic.foo();
//        JmvStatic.bar();
        JmvStatic.Companion.foo();
        JmvStatic.Companion.bar();
        try {
            JavaAnnotationsKt.testThrows();
        } catch (IOException e) {
            System.out.println("IOException");
        }
        JavaAnnotationsKt.testNotNull(null);
    }

}

File: src/main/kotlin/org/avlasov/example/other/java/JavaAnnotations.kt
package org.avlasov.example.other.java

import java.io.IOException
import java.security.Provider

/**
 *   Created By artemvlasov on 2018-10-30
 **/
class Key(val value: Int) {
    companion object {
        @JvmField val COMPARATOR: Comparator<Key> = compareBy { it.value }
    }
}

object Singleton {
    lateinit var provider: Provider
}

object Obj {
    const val CONST = 1
}

class C {
    companion object {
        const val VERSION = 9
    }
}

const val MAX = 239

class JmvStatic {
    companion object {
        @JvmStatic fun foo() {}
        fun bar() {}
    }
}

@Throws(IOException::class)
fun testThrows() {
    throw IOException()
}

fun testNotNull(str: String) = str.length

